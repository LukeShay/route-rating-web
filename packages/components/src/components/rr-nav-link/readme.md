# rr-nav-link

Navigation link to be used with routerating. This component is used as a child
of `rr-nav-bar`. This component contains a `li` component with a
`stencil-route-link` inside.

<!-- Auto Generated Below -->


## Properties

| Property           | Attribute | Description                                 | Type     | Default     |
| ------------------ | --------- | ------------------------------------------- | -------- | ----------- |
| `url` _(required)_ | `url`     | The relative URL where this should link to. | `string` | `undefined` |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
