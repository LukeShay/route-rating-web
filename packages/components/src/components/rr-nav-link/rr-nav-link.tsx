import { Component, Host, Prop, h } from '@stencil/core'

@Component({ tag: 'rr-nav-link' })
export class RrNavLink {
  /**
   * The relative URL where this should link to.
   */
  @Prop() url!: string

  render() {
    return (
      <Host>
        <li>
          <stencil-route-link url={this.url} activeClass="active" exact={true}>
            <slot />
          </stencil-route-link>
        </li>
      </Host>
    )
  }
}
