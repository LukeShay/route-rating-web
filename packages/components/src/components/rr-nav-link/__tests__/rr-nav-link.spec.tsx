import { RrNavLink } from '../rr-nav-link'
import { newSpecPage } from '@stencil/core/testing'

describe('rr-nav-link', () => {
  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [RrNavLink],
      html: `<rr-nav-link url="/some/url">Some URL<button>A random button for testing</button></rr-nav-link>`,
    })

    expect(page.root).toMatchSnapshot()
  })
})
