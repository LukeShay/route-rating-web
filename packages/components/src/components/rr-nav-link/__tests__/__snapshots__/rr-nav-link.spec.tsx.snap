// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rr-nav-link should match snapshot 1`] = `
<rr-nav-link url="/some/url">
  <!---->
  <li>
    <stencil-route-link activeclass="active" exact="" url="/some/url">
      Some URL
      <button>
        A random button for testing
      </button>
      &lt;button&gt;A random button for testing&lt;/button&gt;
    </stencil-route-link>
  </li>
</rr-nav-link>
`;
